modules/file-rotation/file-rotation.c

```diff
@@ -22,29 +22,46 @@
  */
 
 #include "file-rotation.h"
+#include "modules/http/http-signals.h"
+#include "modules/affile/file-signals.h"
 #include "driver.h"
 
-struct _FileRotationPlugin
+static void
+_slot_append_file_rotation_request(FileRotationPlugin *self, FileFlushSignalData *data)
 {
-  LogDriverPlugin super;
-  gsize size;
-};
+
+  g_string_append_printf(self->filename, "%s", data->filename->str);
+}
 
 static gboolean
 file_rotation_attach_to_driver(LogDriverPlugin *s, LogDriver *driver)
 {
-  msg_debug("File rotation plugin has been attached", evt_tag_str("to", driver->id));
+  SignalSlotConnector *ssc = driver->super.signal_slot_connector;
+
+  msg_debug("File rotation plugin has been attached", evt_tag_str("to", s->name));
```

Comment:
The original `driver->id` was correct in my opinion, it showed the driver we attached our file-rotation plugin to.

Reply:
updated